REFERENCE.md

```diff
@@ -0,0 +1,17 @@
+[//]: # (SPDX-License-Identifier: CC-BY-4.0)
+
+Relation to reference implementation
+====================================
+
+mlkem-native is a fork of the ML-KEM [reference implementation](https://github.com/pq-crystals/kyber).
+
+The following gives an overview of the major changes:
```

Comment:
Two things to add here maybe: 
- more inlining: We moved the modular multiplications/reductions into headers. This is probably the largest contributor to better performance
- change to KeccakX4 in appropriate places to faciliate using batched Keccak implementations.

Reply:
> This is probably the largest contributor to better performance 

Do we know that?

Reply:
Looking at the [no-opt performance graphs](https://pq-code-package.github.io/mlkem-native/dev/bench/) the largest performance bump in performance was when link-time optimization was enabled. I remember that afterwards you made various changes to reduce the reliance on lto. 
My memories were that this was vastly due to inlining modular multiplications. 
But this cannot be shown with our benchmarks as we do enable lto. One would have to benchmark without lto to find out. Let's not mention any specifics regarding performance here.